nixos/modules/services/web-apps/timetagger.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkEnableOption mkIf mkOption types literalExpression;
+
+  cfg = config.services.timetagger;
+in {
+
+  options = {
+    services.timetagger = {
+      enable = mkEnableOption ''
+        Tag your time, get the insight
+
+        Note: This app does not do authentication.
+        You must setup authentication yourself or run it in an environment where
+        only allowed users have access.
+      '';
+
+      bindAddr = mkOption {
+        description = "Address to bind to";
+        type = types.str;
+
+        example = "127.0.0.1";
+        default = "127.0.0.1";
+      };
+
+      port = mkOption {
+        description = "Port to bind to";
+        type = types.int;
+
+        example = 8080;
+        default = 8080;
+      };
+
+      package = mkOption {
+        description = ''
+          Use own package for starting timetagger web application.
+
+          The ${literalExpression ''pkgs.timetagger''} package only provides a
+          "run.py" script for the actual package
+          ${literalExpression ''pkgs.python3Packages.timetagger''}.
+
+          If you want to provide a "run.py" script for starting timetagger
+          yourself, you can do so with this option.
+        '';
+
+        default = null;
+        type = types.package;
+      };
+    };
+  };
+
+  config = let
+    timetaggerPkg = if !isNull cfg.package then cfg.package else
+      pkgs.timetagger.overwriteAttrs {
+        addr = cfg.bindAddr;
+        port = cfg.port;
+      };
```

Comment:
If you take my advice above with setting the `default` in `package` then this can be dropped.